### PR TITLE
Add Verified and Top Project Badge

### PR DESCRIPTION
**Pull request by prachigarg19**
Issue: 
Changes in this pull request:
- Given the `approval` flag has been set in the backend, when a user makes a donation, then the blue check mark is shown behind the project name.
- Given the `Top Project` flag has also been set in the backend, when a user makes a donation, then they will see a label in the upper right corner of the image.
-

@

### CODE REVIEW

File: src/Donations/index.tsx [hunk 1]
```diff
@@ -176,6 +177,7 @@ function DonationInfo() {
           <BackButton color={"#000"} />
         </button>
       )}
+      {router.query.to && <div className={"topProjectBadge"}>Top Project</div>}
```
Comment (norbertschuler) [inline src/Donations/index.tsx:180]: I see this now for every project. I assume this is also still a draft?

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/Donations/index.tsx (hunk 1 above)
Comment (prachigarg19) [reply src/Donations/index.tsx:180]: Yes, both of the PRs are in draft as the work in backend is still in progress.

Comment (norbertschuler) [reply src/Donations/index.tsx:180]: Converted both to draft pull requests (as provided as a feature by GitHub).

**Comment by mariahosfeld** [comment]
I had a quick look, for the project where I would expect a is Approved/Verified badge, I see nothing, even though it is correctly exposed in the API.

**Comment by norbertschuler** [comment]
> I had a quick look, for the project where I would expect a is Approved/Verified badge, I see nothing, even though it is correctly exposed in the API.

Me too. https://planet-webapp-o8aj7p5kt-planetapp.vercel.app/jaguarsiembra shows a badge, but https://donate-with-planet-l43g4p21h-planetapp.vercel.app/?to=jaguarsiembra&step=donate does not - both should use the same staging backend and data.

**Comment by mariahosfeld** [comment]
It now shows the badges for all conditions.

**Review by sunilsabatp — APPROVED** [review]

**Comment by sunilsabatp** [comment]
when I search for  yucatan and visited that project in planet donation I  saw the top project batch  but when I clicked back button in the browser to reach the landing page of planet donation. the top project batch still exists. 
image

**Comment by mariahosfeld** [comment]
Functionality of the change works for me.

**Review by norbertschuler — APPROVED** [review]
Works for me. I just wondered if we want to have the link to pp.eco clickable later on?

Bildschirm­foto 2022-11-19 um 11 18 38

File: public/locales/de/common.json [hunk 2]
```diff
@@ -175,5 +175,8 @@
   "planetCashTC": "Indem ich auf \"PlanetCash-Konto erstellen\" klicke, erkläre ich mich mit den <0>Allgemeinen Geschäftsbedingungen</0> der Plant-for-the-Planet-Plattform und von PlanetCash einverstanden.",
   "planetCashIUnderstand": "Ich verstehe, dass PlanetCash Spenden/Zahlungen nicht zurückerstattet werden können, und dass nicht zugewiesene Zahlungen nach sechs Monaten automatisch Projekten auf der Plattform zugewiesen werden.",
   "createPlanetCashAccount": "PlanetCash Konto erstellen",
+  "myContributionIsPartOfThat": "Und meine Spende ist Teil davon.",
+  "verifiedIconInfo": "Dieses Projekt wurde in einer mehrtägigen Feldstudie überprüft. Um mehr zu erfahren, besuchen Sie das Projekt auf pp.eco."
+}
```
Comment (norbertschuler) [inline public/locales/de/common.json:180]: You broke this file, so translation won't work any more!

File: public/locales/en/common.json [hunk 3]
```diff
@@ -175,5 +175,8 @@
   "planetCashTC": "By clicking \"Create PlanetCash Account\", I agree to the <0>Terms and Conditions</0> of Plant-for-the-Planet Platform, and PlanetCash.",
   "planetCashIUnderstand": "I understand that PlanetCash donation/payments are non-refundable, and unallocated payments will be auto allocated to projects on the Platform after six months of Payment.",
   "createPlanetCashAccount": "Create PlanetCash Account",
+  "myContributionIsPartOfThat": "And my contribution is a part of that.",
+  "verifiedIconInfo": "This project has been verified by a multi day field review. To learn more visit the project on pp.eco."
+}
```
Comment (norbertschuler) [inline public/locales/en/common.json:180]: You broke this file as well!

**Review by norbertschuler — CHANGES_REQUESTED** [review]

**Review by norbertschuler — APPROVED** [review]
Works for me. The demo project https://donate-with-planet-6kbvpidrp-planetapp.vercel.app/?to=jaguarsiembra&step=donate is no more showing the badge probably because staging backend got overwritten with production data.

**Comment by mariahosfeld** [comment]
Use the same style of approved badge as on the WebApp.

**Comment by mariahosfeld** [comment]
Styles work now. Please check the alignment of the verified badge and adjust, so that it looks like it does on the WebApp.

**Comment by mariahosfeld** [comment]
The info text is now displayed in one single line, change the layout back to having breaks making it more compact.

**Comment by mariahosfeld** [comment]
Please make sure that the width is on a normal desktop screen at least 3x the width of what it is now (see Screenshot 1). If checked on a mobile view, the width works (see Screenshot 2)
Screenshot 1:
grafik
Screenshot 2:
grafik

**Comment by mariahosfeld** [comment]
Tested with both Edge and Firefox, the width issue still persists:
Screenshot Edge:
grafik
Screenshot Firefox:
grafik

**Comment by mariahosfeld** [comment]
Layout looks good now, checked on Firefox and Edge.

File: package.json
Comment (mohitb35) [inline package.json:1]: @prachigarg19 

1. "@material-ui/core": "^4.11.3",  "@material-ui/lab": "^4.0.0-alpha.57" --> these are the old mui libraries and have been removed. Please be careful while handling merge conflicts - if you're not sure about why something was removed, please ask the team.

2. Why have "immutable": "^4.3.0" and "tiny-warning": "^1.0.3" been added? I do not see where they are used in the project

File: src/Donations/index.tsx [hunk 4]
```diff
@@ -19,6 +19,21 @@ import {
 } from "src/Utils/donationStepConstants";
 import { useRouter } from "next/router";
 import BackButton from "public/assets/icons/BackButton";
+import VerifiedIcon from "@mui/icons-material/Verified";
+import Popover from "@material-ui/core/Popover";
+import Typography from "@material-ui/core/Typography";
+import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
```
Comment (mohitb35) [inline src/Donations/index.tsx:25]: Please use the current mui dependencies. You will need to update the code, and also check that functionality/UX is as expected.

File: package.json
Comment (mohitb35) [inline package.json:1]: @prachigarg19

1. "@material-ui/core": "^4.11.3", "@material-ui/lab": "^4.0.0-alpha.57" --> these are the old mui libraries and have been removed while updating to MUI 5 and next 13. Places where these are still used will need to be updated. **Please be careful while handling merge conflicts - if you're not sure about some changes, please ask the team.**

2. Why have "immutable": "^4.3.0" and "tiny-warning": "^1.0.3" been added? I do not see where they are used in the project.

If you've edited the dependencies in package.json directly, that should be avoided. Use npm install package-name instead.

File: src/Donations/index.tsx (hunk 4 above)
Comment (mohitb35) [inline src/Donations/index.tsx:25]: "@material-ui/core" has been replaced with "@mui/material" during the next 13 upgrade, as it is not compatible with react 18.  Please use the current MUI dependencies. You will need to update the code, and also check that functionality/UX is as expected.

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@prachi, please see my comments

**Comment by mariahosfeld** [comment]
Closed in favor of #407

**Pull request closed without merging**